src/Entity/Server.php

```diff
@@ -397,6 +401,47 @@ protected function getErrorHandler() {
     };
   }
 
+  /**
+   * Logs result errors if any.
+   *
+   * @param \GraphQL\Server\OperationParams $operation
+   * @param \Drupal\graphql\GraphQL\Execution\ExecutionResult $result
+   */
+  protected function logErrors(OperationParams $operation, CacheableExecutionResult $result): void {
+    if (empty($result->errors)) {
+      return;
+    }
+
+    $unsafeErrors = array_filter($result->errors, function ($e) {
+      return !($e instanceof ClientAware) || !$e->isClientSafe();
+    });
+    $isPreviousErrorLogged = FALSE;
+    foreach ($unsafeErrors as $error) {
+      if ($error->getPrevious() instanceof \Throwable) {
+        _drupal_log_error(ErrorUtil::decodeException($error->getPrevious()));
+        $isPreviousErrorLogged = TRUE;
+      }
+    }
+
+    if ($unsafeErrors) {
+      \Drupal::logger('graphql')->error(
+        "There were errors during a GraphQL execution.\n{see_previous}\nDebug:\n<pre>\n{debug}\n</pre>",
+        [
+          'see_previous' => $isPreviousErrorLogged ? 'See the previous log messages for the error details.' : '',
```

Comment:
I'm personally never a fan of the "see the previous log messages" because on busy sites they're hardly ever next to each other. Would there be a good way to get those errors all into a single error log message?

On systems that send alerts based on the amount of errors this can also easily trip monitoring for a single request, which may not be desirable (e.g. if a single request caused the same error to happen multple times).